### PR TITLE
Snapshots for actions (closes #4779)

### LINKED ISSUE #4779: Provide a snapshot of the target element initial state in reportTestActionDone
This feature should be optional and be enabled only if a reporter that supports it is specified.

Part of https://github.com/DevExpress/testcafe/issues/4745.

### PR DESCRIPTION
**Pull request by AlexKamaev**
This PR implements snapshots for actions.
At this moment snapshot for action has the following format:
```JS
snapshot = {
    tagName: 'span',
    attributes: {
        id: 'id',
        className: 'className',
        ....
    }
}
```
Besides, I suppose to change current reporter API. Now the `reportTestActionDone` methods has the `command` argument which looks this way:
```JS
command: {
    selector: 'Selector(\'#target\')',
    type:     'right-click'
},
```
I provide the following approach: 
```JS
command: {
    type: 'click'
    selector: {
        query: 'Selector(\'#target\')',
        element: {
            tagName: 'span',
            attributes: {
            id: 'id',
            className: 'className'
        }
    }
},
```
This is not a BC since the `reportTestActionDone` is not documented.

**I will fix the tests after API is confirmed.**

### CODE REVIEW

**Comment by AndreyBelym** [comment]
I like the API. The only thing that `selector.expression` could be a better fit here. `query` is a process, an operation while `expression` is a code. Definitely needs @VasilyStrelyaev for this.

**Comment by VasilyStrelyaev** [comment]
Yeah, `expression` is good, thanks, @AndreyBelym

**Comment by AlexKamaev** [comment]
FPR

File: src/reporter/format-command.ts [hunk 1]
```diff
@@ -0,0 +1,133 @@
+import { ExecuteSelectorCommand, ExecuteClientFunctionCommand } from '../test-run/commands/observation';
+import { NavigateToCommand, SetNativeDialogHandlerCommand, UseRoleCommand } from '../test-run/commands/actions';
+import { createReplicator, SelectorNodeTransform } from '../client-functions/replicator';
+
+interface Command {
```
Comment (miherlosev) [inline src/reporter/format-command.ts]: I propose to split the code from this file into 3 files:
```js
command/interfaces.ts
command/command-report-item.ts
command/format-command.ts
```

**Pull request merged**